core_lib/src/interface/scribblearea.cpp

```diff
@@ -1119,18 +1126,19 @@ void ScribbleArea::prepCanvas(int frame, QRect rect)
     o.nNextOnionSkinCount = mPrefs->getInt(SETTING::ONION_NEXT_FRAMES_NUM);
     o.fOnionSkinMaxOpacity = mPrefs->getInt(SETTING::ONION_MAX_OPACITY);
     o.fOnionSkinMinOpacity = mPrefs->getInt(SETTING::ONION_MIN_OPACITY);
-    o.bAntiAlias = mPrefs->isOn(SETTING::ANTIALIAS);
-    o.bGrid = mPrefs->isOn(SETTING::GRID);
-    o.nGridSizeW = mPrefs->getInt(SETTING::GRID_SIZE_W);
-    o.nGridSizeH = mPrefs->getInt(SETTING::GRID_SIZE_H);
-    o.bAxis = false;
-    o.bThinLines = mPrefs->isOn(SETTING::INVISIBLE_LINES);
-    o.bOutlines = mPrefs->isOn(SETTING::OUTLINES);
-    o.nShowAllLayers = mShowAllLayers;
-    o.bIsOnionAbsolute = (mPrefs->getString(SETTING::ONION_TYPE) == "absolute");
-    o.scaling = mEditor->view()->scaling();
-    o.onionWhilePlayback = mPrefs->getInt(SETTING::ONION_WHILE_PLAYBACK);
-    o.isPlaying = mEditor->playback()->isPlaying() ? true : false;
+    o.bAntiAlias           = mPrefs->isOn(SETTING::ANTIALIAS);
+    o.bGrid                = mPrefs->isOn(SETTING::GRID);
+    o.nGridSizeW           = mPrefs->getInt(SETTING::GRID_SIZE_W);
+    o.nGridSizeH           = mPrefs->getInt(SETTING::GRID_SIZE_H);
+    o.bAxis                = false;
+    o.bThinLines           = mPrefs->isOn(SETTING::INVISIBLE_LINES);
+    o.bOutlines            = mPrefs->isOn(SETTING::OUTLINES);
+    o.eLayerVisibility     = mLayerVisibility;
+    o.fLayerVisibilityThreshold = mPrefs->getFloat(SETTING::LAYER_VISIBILITY_THRESHOLD);
```

Comment:
Well now you've got to realign everything with this :yum:

Reply:
It wasn't aligned before the pull request though, so I say let's leave it like that and do a proper refactoring later.